src/main/java/cloud/fogbow/ras/core/SynchronizationManager.java

```diff
@@ -0,0 +1,40 @@
+package cloud.fogbow.ras.core;
+
+public class SynchronizationManager {
+    private static SynchronizationManager instance;
+    private boolean reloading;
+    private ProcessorsThreadController processorsThreadController;
+
+    private SynchronizationManager() {
+        this.reloading = false;
+    }
+
+    public static synchronized SynchronizationManager getInstance() {
+        if (instance == null) {
+            instance = new SynchronizationManager();
+        }
+        return instance;
+    }
+
+    public void setProcessorsThreadController(ProcessorsThreadController processorsThreadController) {
+        this.processorsThreadController = processorsThreadController;
+    }
+
+    public boolean isReloading() {
+        return this.reloading;
+    }
+
+    public void reload() {
+        this.reloading = true;
+        while (ApplicationFacade.getInstance().getOnGoingRequests() != 0)
```

Comment:
I wonder if it is possible to implement this without busy wait. Depending on how fast the API operations are performed, maybe it is worth to use semaphores here. Cannot think of any good solution using semaphores though.